config-generator/src/main/resources/infinispan.xml

```diff
@@ -48,6 +48,19 @@
                     <gsp:scriptlet>
                         }
                     </gsp:scriptlet>
+                    <gsp:scriptlet>
+                        <![CDATA[
+                        if (keystore && !keystore.path && keystore.selfSignCert) {
+                        ]]>
+                    </gsp:scriptlet>
+                    <server-identities>
+                        <ssl>
+                            <keystore path="${keystore.crtPath}/selfsigned_keystore.p12" alias="server" keystore-password="password" generate-self-signed-certificate-host="localhost"/>
+                        </ssl>
+                    </server-identities>
+                    <gsp:scriptlet>
+                        }
+                    </gsp:scriptlet>
```

Comment:
I don't think we should add another if branch in `infinispan.xml`. Instead we should re-use the existing one and surround the `generate-self-signed-certificate-host` attribute with a gsp tag.

For this to work, we should add a `keystore.selfSignCert = false` entry to `default-configs.yaml` and set `path` and `password` in the `ConfigGenerator::configureKeystore` method.

Reply:
@ryanemerson any quick hint on how to add the `generate-self-signed-...` conditionally? I've tried some syntax, but no luck.

Reply:
You could use a `<gsp:expression>`? http://docs.groovy-lang.org/docs/next/html/documentation/template-engines.html#_xmltemplateengine

Reply:
yeah good hint @ryanemerson , thank you!